src/main/java/picard/cmdline/argumentcollections/ReferenceArgumentCollection.java

```diff
@@ -25,13 +25,55 @@
 package picard.cmdline.argumentcollections;
 
 import java.io.File;
+import java.nio.file.Path;
+
+import htsjdk.samtools.util.Log;
+import picard.nio.PicardHtsPath;
 
 /**
  * Base interface for a reference argument collection.
  */
 public interface ReferenceArgumentCollection {
     /**
-     * @return The reference provided by the user, if any, or the default, if any.
+     * @return The reference provided by the user, if any, or the default, if any, as a File. May be null.
      */
     File getReferenceFile();
+
+    /**
+     * @return The reference provided by the user, if any, or the default, if any, as an nio Path. May be null.
+     */
+    default Path getReferencePath() { return getReferenceFile() == null ? null : getReferenceFile().toPath(); }
+
+    /**
+     * @return The reference provided by the user, if any, or the default, if any, as a PicardHtsPath. May be null.
+     */
+    default PicardHtsPath getHtsPath() {
+        return getReferenceFile() == null ?
+                null :
+                new PicardHtsPath(getReferenceFile().getAbsolutePath()); }
+
+    /**
+     * @return A "safe" way to obtain a File object for any reference path.
+     *
+     * For files that reside on a local file system, this returns a valid File object. Files that reside on
+     * a non-local file system can't be accessed via a File object, so return a placeholder File object that
+     * will defer failure when/if the file is actually accessed. This allows code paths that blindly propagate
+     * the value returned by calls to getReferenceFile to not get an NPE, and to fail gracefully downstream
+     * with an error message that includes the reference file specifier.
+     */
+    static File getFileSafe(final PicardHtsPath picardPath, final Log log) {
+        if (picardPath == null) {
+            return null;
+        } else if (picardPath.getScheme().equals(PicardHtsPath.FILE_SCHEME)) {
+            // file on a local file system
+            return picardPath == null ? null : picardPath.toPath().toFile();
```

Comment:
`return picardPath.toPath().toFile()` because `picardPath` is never null if the code gets here